client_core/src/com/mygdx/game/client_core/model/GameState.java

```diff
@@ -21,14 +24,18 @@ public class GameState {
   private final ComponentMapper<Coordinates> coordinatesMapper;
   private final ComponentMapper<SubField> subFieldMapper;
   private final ComponentMapper<Score> scoreMapper;
+  private final ComponentMapper<Movable> movableMapper;
   private final Map<Coordinates, IntArray> entitiesAtCoordinateGame;
+  private final List<Integer> technologies;
```

Comment:
IntArray

Reply:
Tutaj nie mogłem dać IntArray, bo on domyślnie zmienia rozmiar do potęgi dwójki, i elementy nieustawione są zerami, co mi w pewnm momencie przeszkadzało

Reply:
To może te 0 jednak mogą oznaczać brak entity? Trzeba to sprawdzić, jakie id ma pierwsze zrobione entity.

Reply:
pierwsze zrobione entity ma id 0

Reply:
możesz mi później pokazać czemu to był problem?

Reply:
był problem bo robiła się taka lista <id, id, 0, 0, 0, 0, 0, 0, ...> no i brało mi przez to entity które nie były tymi które chciałem

Reply:
zmieniłem teraz to na IntArray i działa normalnie 
```java
    int size = technologies.getAllTechnologies().size;
    for (int i = 0; i < size; i++) {
      var entityId = technologies.getAllTechnologies().get(i);
```
getTechnologies zwraca mi tego IntArray